### PR TITLE
(fix) Issues with search / filters

### LINKED ISSUE #440: UI/UX: issues with search/filters on the Positions and Conditions pages
Related to #406, #407, #425

**Conditions page:** 
1. **Number of outcomes filter**: it would be nice to add a FE logic for 
- **filling in the fields with outcomes in range [2-256]**

- **when decimals and negative amounts are typed into the fields**

2. **It would be nice to disable autofill option for the input fields in the Number of outcomes filter**

**Conditions and Position pages:** 

3. **It would be nice to show a tooltip or to reduce the font size of the text in the search option.** 

-  **It is a bit difficult to remove a value from a date field once it has been filled in** (a user should remove a date, then a month and then a year)

4. It would be nice to remove or disable a cross icon in the search field when the field is empty

### PR DESCRIPTION
**Pull request by gabitoesmiapodo**
Closes #440 

- _Filling in the fields with outcomes in range..._ -> Min / max range is there now, but only if the user uses the keyboard's arrows to fill the numbers. They can write numbers out of range. `(*)`
- _When decimals and negative amounts are typed into the fields..._ -> It shouldn't be possible to enter a minus / dash symbol or a period symbol. `(**)`
- _It would be nice to disable autofill option for the input fields in the Number of outcomes filter..._ -> Done.
- _It would be nice to show a tooltip or to reduce the font size of the text in the search option._ -> User can check the dropdown options or the input's placeholder to see what's selected. However, I modified the styles so the button will expand depending on the selected option... unless there's a extremely long text (as with "Wrapped Collateral Address") in which case the text will be cut because I can't let the button to expand indefinitely and use all of the input field's space. It'll be fine for most options.
- _It is a bit difficult to remove a value from a date field once it has been filled..._ -> This is more involved than it seems. I created an issue here https://github.com/gnosis/conditional-tokens-explorer/issues/531
- _It would be nice to remove or disable a cross icon in the search field when the field is empty..._ -> Done.

`(*)` The user could still enter whatever number the want to enter by writing it. Logic to avoid this could be too convoluted and unnecessary TBH, and ranges are checked when filters are applied so checking this on the frontend too shouldn't be needed. Minimum will be 2 and maximum will be 256 regardless.
`(*)` and `(**)` The user could paste a wrong value (negative number, decimal number, out of range values, etc.), and maybe do something else to avoid these validations. However: values are (or should be) validated internally and - as long as we do that- it shouldn't really matter.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr532--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr532--conditionaltokensexplorer.review.gnosisdev.com

File: src/components/filters/MinMaxFilter/index.tsx [hunk 1]
```diff
@@ -87,15 +88,27 @@ export const MinMaxFilter: React.FC<Props> = (props) => {
...
+    if (min) {
+      setMin(Math.trunc(min))
+      if (min < MIN_OUTCOMES_ALLOWED) setMin(MIN_OUTCOMES_ALLOWED)
+    }
+
+    if (max) {
+      setMax(Math.trunc(max))
+      if (max > MAX_OUTCOMES_ALLOWED) setMax(MAX_OUTCOMES_ALLOWED)
+    }
+  }, [max, min])
```
Comment (hpmaxi) [inline src/components/filters/MinMaxFilter/index.tsx:101]: For this logic we should take into account that setState operations are batched, so if max was `5`  and changes to `2.5` the callback is executed in this way

```
setMin(Math.trunc(2.5))
if(2.5 < MIN_OUTCOMES_ALLOWED) setMin(MIN_OUTCOMES_ALLOWED)
``` 
I guess this run again, but maybe is clearer if you only make one set
```
const max = Math.trunc(max) // or maxTruncated
setMax(max > MAX_OUTCOMES_ALLOWED ? MAX_OUTCOMES_ALLOWED : max)
```

Anyways it works in the way it is right now.

**Review by hpmaxi — APPROVED** [review]

File: src/components/filters/MinMaxFilter/index.tsx (hunk 1 above)
Comment (gabitoesmiapodo) [reply src/components/filters/MinMaxFilter/index.tsx]: Yeah, I like your solution more. Problem is: `min` or `more` could be `null` so I end up with something like this to make the compiler happy:

```
const minTrunc = min !== null ? Math.trunc(min) : null
const maxTrunc = max !== null ? Math.trunc(max) : null

setMin(minTrunc !== null && minTrunc < MIN_OUTCOMES_ALLOWED ? MIN_OUTCOMES_ALLOWED : minTrunc)
setMax(maxTrunc !== null && maxTrunc > MAX_OUTCOMES_ALLOWED ? MAX_OUTCOMES_ALLOWED : maxTrunc)
```

Not sure if there's a simpler solution.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr532--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr532--conditionaltokensexplorer.review.gnosisdev.com

File: src/components/filters/MinMaxFilter/index.tsx (hunk 1 above)
Comment (hpmaxi) [reply src/components/filters/MinMaxFilter/index.tsx]: You could use a short-circuit &&, and do something like `min && Math.trunc(min)`. 0 is considered falsy but 0 truncated is 0, in my opinion, it's ok in the last version.

**Review by mariano-aguero — APPROVED** [review]

**Comment by elena-zh** [comment]
@gabitoesmiapodo , I have reviewed the changes. Everything looks good to me. 
The only thing I can mention here is the issue with filling in the outcomes fields with the range [2-256]. I reported the issue based on the Outcomes field behavior on the Prepare Condition field. As you can see, user is warned about the min and max outcomes range there. 
But if it does not makes sense for the filter, it is no problem =)

**Comment by gabitoesmiapodo** [comment]
@elena-zh I'd say that in this case the user could enter an out of range value and the filter would work anyways within the valid range, so it's not really important to let them know what the "problem" is.

For every other form, if we don't tell them what's going on they would need to guess, or they could end up frustrated and stop trying.

**Pull request merged**